libraries/src/MVC/Model/ListModel.php

```diff
@@ -199,17 +207,14 @@ public function getIsEmptyState(): bool
 	 */
 	protected function _getListQuery()
 	{
-		// Capture the last store id used.
-		static $lastStoreId;
-
 		// Compute the current store id.
 		$currentStoreId = $this->getStoreId();
 
 		// If the last store id is different from the current, refresh the query.
-		if ($lastStoreId != $currentStoreId || empty($this->query))
+		if ($this->lastQueryStoreId != $currentStoreId || empty($this->query))
```

Comment:
```suggestion
		if ($this->lastQueryStoreId !== $currentStoreId || empty($this->query))
```
Just changed this to trigger drone, but should be type safe anyway